homeassistant/components/template/switch.py

```diff
@@ -134,22 +200,21 @@ class SwitchTemplate(TemplateEntity, SwitchEntity, RestoreEntity):
     def __init__(
         self,
         hass: HomeAssistant,
-        object_id,
         config: ConfigType,
-        unique_id,
+        unique_id: str | None,
     ) -> None:
         """Initialize the Template switch."""
-        super().__init__(
-            hass, config=config, fallback_name=object_id, unique_id=unique_id
-        )
-        if object_id is not None:
+        if (object_id := config.get(CONF_OBJECT_ID)) is not None:
             self.entity_id = async_generate_entity_id(
                 ENTITY_ID_FORMAT, object_id, hass=hass
             )
+        super().__init__(
+            hass, config=config, fallback_name=object_id, unique_id=unique_id
+        )
```

Comment:
Why is the super call moved?

Reply:
Super call is removed for the fallback name to suggest the entity_id with the old style configuration.  However, I made that assumption.  Looking at switch, it seems it's not needed in that order.  I'll make the change.